impl/src/main/java/com/okta/sdk/impl/util/OAuth2Utils.java

```diff
@@ -0,0 +1,163 @@
+
+        String jwt = Jwts.builder()
+            .setAudience(clientConfiguration.getBaseUrl() + "/oauth2/default/v1/token")
+            .setIssuedAt(Date.from(now))
+            .setExpiration(Date.from(now.plus(1L, ChronoUnit.HOURS)))
+            .setIssuer(clientId)
+            .setSubject(clientId)
+            .claim("jti", UUID.randomUUID().toString())
+            .signWith(signatureAlgorithm, privateKey)
+            .compact();
+
+        return jwt;
+    }
+
+    public static String getOAuth2AccessToken(ClientConfiguration clientConfiguration)
+        throws HttpException, IOException, InvalidKeySpecException, NoSuchAlgorithmException {
+        String signedJwt = createSignedJWT(clientConfiguration);
+
+        RequestExecutor requestExecutor = okHttpRequestExecutorFactory.create(clientConfiguration);
+
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.add("Accept", "application/json");
+        httpHeaders.add("Content-Type", "application/x-www-form-urlencoded");
+
+        QueryString queryString = new QueryString();
+        queryString.put("grant_type", "client_credentials");
+        queryString.put("scope", clientConfiguration.getScopes());
+        queryString.put("client_assertion_type",
+            URLEncoder.encode("urn:ietf:params:oauth:client-assertion-type:jwt-bearer", "UTF-8"));
+        queryString.put("client_assertion", signedJwt);
+
+        DefaultRequest accessTokenRequest =
+            new DefaultRequest(HttpMethod.POST, "/oauth2/v1/token", queryString, httpHeaders);
+        Response accessTokenResponse = requestExecutor.executeRequest(accessTokenRequest);
+
+        if (accessTokenResponse != null && accessTokenResponse.getBody() != null) {
+            Reader reader = new InputStreamReader(accessTokenResponse.getBody(), StandardCharsets.UTF_8);
+            Type mapType = new TypeToken<Map<String, String>>(){}.getType();
+            Map<String, String> responseMap = gson.fromJson(reader, mapType);
+
+            if (accessTokenResponse.getHttpStatus() == HttpURLConnection.HTTP_OK) {
+                return responseMap.get("access_token");
+            }
+            else {
+                String error = responseMap.get("error");
+                String error_description = responseMap.get("error_description");
+                if (error != null && error.length() > 0 &&
+                    error_description != null && error_description.length() > 0) {
+                    throw new HttpException("Received HTTP " + accessTokenResponse.getHttpStatus() +
+                        " response from Authorization Server. error - " + error +
+                        ", error_description - " + error_description);
+                }
+                else {
+                    throw new HttpException("Received HTTP " + accessTokenResponse.getHttpStatus() +
+                        " response from Authorization Server");
+                }
+            }
```

Comment:
Do something like this instead:
https://github.com/okta/okta-sdk-java#call-other-api-endpoints

Exceptions will automatically be mapped as well

Reply:
Noted